### PR TITLE
[PIM-6127] In the family import, the attributes required should be in the family

### PR DESCRIPTION
**Pull request by pierallard**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

Actually, you can import a family with a required attribute it is NOT in the attributes list of the family.
This PR fixes this wrong behavior and skip this item with a warning.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | y
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: features/import/import_families.feature [hunk 1]
```diff
@@ -78,3 +78,21 @@ Feature: Import families
...
+      boots;Boots;sku,name,manufacturer,description,weather_conditions,price,rating,side_view,top_view,size,color,lace_color;name;sku,name,description,weather_conditions,price,rating,side_view,size,color;sku,name,price,size,color
+      wrong_family;Wrong Family;sku,name;name;description;description
+      """
+    And the following job "csv_footwear_family_import" configuration:
+      | filePath | %file to import% |
+    When I am on the "csv_footwear_family_import" import job page
+    And I launch the import job
+    And I wait for the "csv_footwear_family_import" job to finish
+    Then I should see the text "Skipped 1"
+    And I should see the text "The attribute \"description\" cannot be an attribute required as it does not belong to the family: Wrong Family"
```
Comment (ahocquard) [inline features/import/import_families.feature]: "a required attribute". And 'to the family "Wrong Family"' (family in double quote for consistency). And we should precise the channel.

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php [hunk 2]
```diff
@@ -18,7 +18,14 @@ class FamilyRequirements extends Constraint
      *
      * @var string
      */
-    public $message = 'Family "%family%" should have requirements for identifier "%id%" and channel(s) "%channels%"';
+    public $messageChannel = 'Family "%family%" should have requirements for identifier "%id%" and channel(s) '.
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php]: necessary to split the line ? you seems to have an automatic rule on that.

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirementsValidator.php [hunk 3]
```diff
@@ -41,19 +45,54 @@ public function __construct(
...
+     * @param FamilyInterface $family
+     * @param Constraint      $constraint
+     */
+    protected function validateRequiredAttributes(FamilyInterface $family, Constraint $constraint)
+    {
+        $familyAttributeCodes = $family->getAttributeCodes();
+
+        $requiredCodes = [];
+        foreach ($family->getAttributeRequirements() as $code => $attributeRequirement) {
+            $requiredCodes[] = $attributeRequirement->getAttributeCode();
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirementsValidator.php]: why not just use : 

```
if (!in_array($attributeRequirement->getAttributeCode(), $familyAttributeCodes) {
     $this->context
                  ->buildViolation($constraint->messageAttribute, ['%attribute%' => $attributeRequirement->getAttributeCode())
                 ->addViolation();
}
```
}

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php [hunk 4]
```diff
@@ -18,7 +18,14 @@ class FamilyRequirements extends Constraint
...
+        '"%channels%"';
+
+    /**
+     * @var string
+     */
+    public $messageAttribute = 'The attribute "%attribute%" cannot be an attribute required as it does not belong '.
+        'to the family';
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php]: Weird, it's not exactly the same message as the behat, where is put the "to the family: Wrong channel" at the end ? 
More over, we should add the channel.
It should be : 
"The attribute "%attribute%" cannot be a required attribute in the channel "%s" as it does not belong to the family "%family%"'

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirementsValidator.php [hunk 5]
```diff
@@ -41,19 +45,54 @@ public function __construct(
...
+        }
+        $requiredCodes = array_unique($requiredCodes);
+
+        foreach (array_diff($requiredCodes, $familyAttributeCodes) as $missingAttributeCode) {
+            $this->context
+                ->buildViolation($constraint->messageAttribute, ['%attribute%' => $missingAttributeCode])
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirementsValidator.php]: %family%", %channel%, see my comment above.

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php (hunk 2 above)
Comment (pierallard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php]: We will see what the specs says :p

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php (hunk 4 above)
Comment (pierallard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirements.php]: The ": family_name" is automatically added at the end of each warning, I have no choice about this.

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirementsValidator.php (hunk 3 above)
Comment (pierallard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyRequirementsValidator.php]: Because it's an ArrayCollection, this not work with in_array

**Pull request merged**